test/integration/targets/apt_key/tasks/apt_key_inline_data.yml

```
@@ -0,0 +1,8 @@
+---
+
+- name: "Ensure import of a deliberately corrupted downloaded GnuPG binary key results in an 'inline data' occurence in the message"
+  apt_key: url=https://github.com/maximmasiutin/ansible/raw/gpg-inline-data-msg-fix/test/integration/targets/apt_key/samples/apt-key-corrupt-zeros-2k.gpg
+  # replace the above URL to the following, after the pull request is accepted
+  # apt_key: url=https://github.com/ansible/ansible/raw/devel/test/integration/targets/apt_key/samples/apt-key-corrupt-zeros-2k.gpg
```

Binary files are best stored in S3 rather than the repository.
```suggestion
  apt_key:
    url: https://ansible-ci-files.s3.us-east-1.amazonaws.com/test/integration/targets/apt_key/apt-key-corrupt-zeros-2k.gpg
```